### PR TITLE
#362 dashboard 3

### LINKED ISSUE #362: Initial dashboard
I would like to have something in place which is more indicative than the current skeleton for the dashboard, without going overboard or creating a full implementation.

Screen Shot 2021-11-03 at 11 00 28 AM

Am basing this off [Dashboard V1](https://zpl.io/VOEAlkk) as I _think_ this was the preferred option ( not certain! )

**Inbound Shipments**
- Show number created today
- Show number created this week ( both could potentially be determined using the list query, but that would be terribly inefficient )

**Outbound Shipments**
- Show number to be picked ( i.e. status === `ALLOCATED` )
- other ideas? The mockup has 'to be picked: This week' but what does that mean?!

**Stock**
- Quantity expired ( quantity? packs? value? )
- Expiring soon ( what is soon? same questions about the unit to use.. quantity, packs, value )

Note: these can be taken inefficiently from the existing API and we'll probably have to do that before the API provides useful information

Unsure what to include under `More Actions` - at this stage perhaps just a link to the list view
I don't believe that we can include any of the 'wizard' buttons right now.. these will require some analysis and then API work

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #362 this is the last part of the basic dashboard I was wanting to do.
This tidies up the layout, which wasn't flexing enough, moves the Shipment widgets to be with their buddies, and adds a couple of stories.

Will make an issue to create a `Stock` list and move the widget alongside it.

### CODE REVIEW

File: packages/common/src/api/OmSupplyApiContext.tsx [hunk 1]
```diff
@@ -60,9 +60,7 @@ export const OmSupplyApiProvider: FC<ApiProviderProps> = ({
   }, [url]);
 
   return (
-    <Provider value={{ api, client, setUrl, setHeader }}>
-      {children}setApi
-    </Provider>
+    <Provider value={{ api, client, setUrl, setHeader }}>{children}</Provider>
```
Comment (mark-prins) [inline packages/common/src/api/OmSupplyApiContext.tsx:63]: looked like an accident, so I removed it 🤷

Comment (mark-prins) [inline packages/common/src/api/OmSupplyApiContext.tsx:63]: looked like an accident, so I removed it 🤷

File: packages/host/src/Host.tsx [hunk 2]
```diff
@@ -197,7 +197,7 @@ const Host: FC = () => {
                           overflow="hidden"
                         >
                           <AppBar />
-                          <Box display="flex" flex={1} overflow="hidden">
+                          <Box display="flex" flex={1} overflow="scroll">
```
Comment (mark-prins) [inline packages/host/src/Host.tsx:200]: didn't seem to damage the other pages - and now if the widgets flow to another row, the scrollbar shows

File: packages/dashboard/src/DashboardService.tsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 import { Grid } from '@openmsupply-client/common';
+import { StockWidget } from './widgets';
 import {
-  InboundShipmentsWidget,
-  OutboundShipmentsWidget,
-  StockWidget,
-} from './widgets';
+  InboundShipmentWidget,
+  OutboundShipmentWidget,
+} from '@openmsupply-client/invoices';
```
Comment (joshxg) [inline packages/dashboard/src/DashboardService.tsx:7]: Maybe `lerna add @openmsupply-client/invoices --scope @openmsupply-client/dashbaord` to be eplicit?

File: packages/common/src/api/OmSupplyApiContext.tsx (hunk 1 above)
Comment (joshxg) [reply packages/common/src/api/OmSupplyApiContext.tsx:63]: 🤣

File: packages/invoices/src/OutboundShipment/Dashboard/Widget.tsx [hunk 4]
```diff
@@ -9,13 +9,13 @@ import {
...
+  Widget,
 } from '@openmsupply-client/common';
-import Widget from './Widget';
-import { StatsPanel } from '../StatsPanel';
 import { useNavigate } from 'react-router';
-import { getOutboundShipmentCountQueryFn } from '../api';
+import { getOutboundShipmentCountQueryFn } from './api';
 
-export const OutboundShipmentsWidget: React.FC = () => {
+export const OutboundShipmentWidget: React.FC = () => {
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/Dashboard/Widget.tsx:18]: Cool! Might actually not be too much work to do the 'mini' dashboards for the top level navigation areas like `/distribution` etc with widgets like these?

**Review by joshxg — COMMENTED** [review]
It's great!!

I just realised I am not totally sure what the 3 columns are grouping - in stock, out stock, stock in general?

File: packages/dashboard/src/DashboardService.tsx (hunk 3 above)
Comment (mark-prins) [reply packages/dashboard/src/DashboardService.tsx:7]: great! thanks

File: packages/invoices/src/OutboundShipment/Dashboard/Widget.tsx (hunk 4 above)
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/Dashboard/Widget.tsx:18]: yeah, should be pretty straightforward.. so maybe that idea is back on the table 😁

**Comment by mark-prins** [comment]
> I just realised I am not totally sure what the 3 columns are grouping - in stock, out stock, stock in general?

Up to us, I suppose. They were in the designs, so I ran with them. So far the in stock is just inbound shipments - could be requisitions & purchase orders as well I guess. We don't have the wizard actions yet, so the panels aren't very specific - i.e. tied to just inbound shipments.

**Pull request merged**